piqture/data_loader/mnist_data_loader.py

```diff
@@ -13,123 +13,130 @@
         img_size (int or tuple[int, int], optional): Size to which images
-        will be resized. Defaults to 28.
-            If integer, images will be resized to a square of that size.
-            If tuple, images will be resized to specified height and width.
-        batch_size (int, optional): Batch size for the dataset.
-        labels (list): List of desired labels.
+            will be resized. Defaults to 28.
+        batch_size_train (int, optional): Batch size for training set. Defaults to 64.
+        batch_size_test (int, optional): Batch size for test set. Defaults to 1000.
+        labels (list, optional): List of desired labels.
         normalize_min (float, optional): Minimum value for normalization.
         normalize_max (float, optional): Maximum value for normalization.
+        split_ratio (float, optional): Ratio to split train/test datasets. Defaults to 0.8.
+        load (str, optional): Indicates whether to load "train", "test", or "both".
+        Defaults to "both".
 
     Returns:
-        Train and Test DataLoader objects.
+        Train and/or Test DataLoader objects, depending on the `load` argument.
     """
-    # Check if img_size is int or tuple.
-    # Also check if tuple entries are int.
+
+    # Validate inputs
     if not isinstance(img_size, (int, tuple)):
-        raise TypeError(
-            "the input img_size must be of the type int or tuple[int, int]."
-        )
+        raise TypeError("img_size must be an int or tuple[int, int].")
 
     if isinstance(img_size, tuple) and not all(
         isinstance(size, int) for size in img_size
     ):
-        raise TypeError("the input img_size must be of the type tuple[int, int].")
-
-    # Check if batch_size is an int.
-    if batch_size:
-        if not isinstance(batch_size, int):
-            raise TypeError("The input batch_size must be of the type int.")
-
-    # Check if labels are a list.
-    if labels:
-        if not isinstance(labels, list):
-            raise TypeError("The input labels must be of the type list.")
-
-    if normalize_max and normalize_min:
-        # Define a custom mnist transforms.
-        mnist_transform = torchvision.transforms.Compose(
-            [
-                torchvision.transforms.ToTensor(),
-                torchvision.transforms.Resize(img_size),
-                MinMaxNormalization(normalize_min, normalize_max),
-            ]
-        )
-    else:
-        # When normalization is not requested; when normalize_min and max are None.
-        mnist_transform = torchvision.transforms.Compose(
-            [
-                torchvision.transforms.ToTensor(),
-                torchvision.transforms.Resize(img_size),
-            ]
-        )
-
-    # Partial application of custom collate function that calls
-    # collate_fn with args.
-    new_batch = []
-    custom_collate = partial(collate_fn, labels=labels, new_batch=new_batch)
+        raise TypeError("img_size tuple must contain integers.")
```

Comment:
Can we revert the TypeError to what it was before? I prefer having more details in the error message.